pandas/_libs/tslibs/period.pyx

```diff
@@ -1567,12 +1567,12 @@ cdef class _Period(object):
 
     @classmethod
     def _maybe_convert_freq(cls, object freq):
+        if not is_offset_object(freq):
```

Comment:
This looks fine, and the timings you posted look nice.  Can you try adding the `not is_offset_object` check at the top of tslibs.offsets.to_offset?  Should get a similar performance improvement in a handful of other places.

Reply:
Sure, will give that a shot. Will be out for a bit so may be several hours before I can post the results.

Reply:
I'm not seeing any significant changes from implementing this - did you have a particular benchmark in mind?

Reply:
I didn’t have any one in mind, but had a hunch.  Thanks for taking a look